src/placeos-models/edge.cr

```diff
@@ -0,0 +1,24 @@
+require "random"
+
+require "./base/model"
+
+module PlaceOS::Model
+  class Edge < ModelBase
+    include RethinkORM::Timestamps
+
+    table :edge
+
+    attribute name : String, es_subfield: "keyword"
+    attribute identity : String = ->{ UUID.random.to_s }
+    attribute secret : String = ->{ Random::Secure.hex(64) }
+
+    ensure_unique :name
```

Comment:
Assuming that names are defined by and intended for use by humans, is it worth ignoring case + stripping whitespace?

Reply:
The `ensure_unique` validator in `rethinkdb-orm` is not ideal at the moment. The current form will mutate the field to the result of whatever transform you apply.
Rethinkdb offers a surprising amount of string manipulations, so we could apply the transform at the db level at the expense of slow ass writes.
We could also add an extra column with a secondary index that we don't serialize and do the unique check on.